### ISSUE TITLE
Replace `protons` with another package

### ISSUE THREAD

**Issue by D4nte** [status-im/js-waku#335]
## Problem

`protons` package is used in several examples and guides and example to encode/decode protobuf messages.

It is now marked as deprecated: https://www.npmjs.com/package/protons

## Solution

Use another package with a simple API.

**Comment by D4nte** [comment]
With protobufjs it's possible to build without bufbuild

https://github.com/filoozom/tfe-uliege-code/tree/main/platform%2Fnode

**Comment by D4nte** [comment]
Maintenance has been resumed :tada: https://github.com/ipfs/protons/commit/74d3b7abf1e857f7320c100734e797855ea728c1

Focusing this issue in the tree-shakeable problem.

**Comment by fryorcraken** [comment]
Protons is facing some performance issues: https://github.com/ipfs/protons/issues/51
The also haven't accepted the PR to align behaviour change with proto3: https://github.com/ipfs/protons/pull/48

When picked up, this issue can focus on ensuring that the protobuf library is tree-shakeable.

**Comment by fryorcraken** [comment]
Check https://buf.build/blog/protobuf-es-the-protocol-buffers-typescript-javascript-runtime-we-all-deserve

**Comment by fryorcraken** [comment]
Need to check if `protons` is tree shakeable also check this convo: https://github.com/ChainSafe/js-libp2p-gossipsub/pull/368#issuecomment-1299609972

**Comment by fryorcraken** [comment]
Do note https://github.com/status-im/status-web/pull/337

**Comment by fryorcraken** [comment]
> Definition from https://github.com/vacp2p/waku are used

Most of the work done here: https://github.com/waku-org/js-waku/pull/1196

Peer exchange still pending: https://github.com/vacp2p/waku/issues/14

**Comment by fryorcraken** [comment]
> Based on #1245, it can be deduced that `protobuf-es` while reducing the total lines of code by 50%, still increases the package size by about 13% which makes `protons` better suited?
> 
> cc @fryorcraken

Ok seems that in terms of bundle size, it does not help. Noting that this issue is specifically about bundle size (tree shakeable code) it would make sense to keep protons.

**Comment by fryorcraken** [comment]
Yes this is a fair idea. But not for 2023 https://notes.status.im/Uz9HeCwZTDSYyOq36Q54cA?edit

I'd suggest to open a different issue and park it.
